nbr/src/main/java/io/nosqlbench/scenarios/simframe/optimizers/findmax/FindmaxFrameFunction.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.scenarios.simframe.optimizers.findmax;
+
+import io.nosqlbench.engine.api.activityapi.core.Activity;
+import io.nosqlbench.engine.api.activityapi.core.RunState;
+import io.nosqlbench.engine.core.lifecycle.scenario.container.ContainerActivitiesController;
+import io.nosqlbench.scenarios.simframe.capture.SimFrameCapture;
+import io.nosqlbench.scenarios.simframe.capture.SimFrameJournal;
+import io.nosqlbench.scenarios.simframe.planning.SimFrameFunction;
+
+public class FindmaxFrameFunction implements SimFrameFunction<FindmaxFrameParams> {
+
+    private final Activity flywheel;
+    private final SimFrameCapture capture;
+    private final SimFrameJournal<FindmaxFrameParams> journal;
+    private final FindmaxConfig settings;
+    private final ContainerActivitiesController controller;
+    private final FindmaxParamModel model;
+
+    public FindmaxFrameFunction(
+        ContainerActivitiesController controller,
+        FindmaxConfig settings,
+        Activity flywheel,
+        SimFrameCapture capture,
+        SimFrameJournal<FindmaxFrameParams> journal,
+        FindmaxParamModel model
+    ) {
+        this.controller = controller;
+        this.settings = settings;
+        this.flywheel = flywheel;
+        this.capture = capture;
+        this.journal = journal;
+        this.model = model;
+    }
+
+    @Override
+    public double value(double[] point) {
+        System.out.println("━".repeat(40));
```

Comment:
#nit

Should these SOPs be converted to actual loggers?

Reply:
That's a good point. They are currently mimicking the behavior of the old findmax but I agree they should probably be converted to logging, along with the other stdout statements in the code.